### ISSUE TITLE
[BUG][MTL] Speaker playback IO error

### ISSUE THREAD

**Issue by aiChaoSONG** [thesofproject/sof#7979]
**Describe the bug**
Playback a wave to speaker with aplay, see io error in terminal immediately. no obviously error in kernel log, and in mtrace, xrun happened.

```
[    1.999690] <inf> host_comp: host_get_copy_bytes_normal: comp:0 0x40000 no bytes to copy, available samples: 384, free_samples: 0
[    2.000706] <inf> host_comp: host_get_copy_bytes_normal: comp:0 0x40000 no bytes to copy, available samples: 384, free_samples: 0

```

**To Reproduce**
simple do aplay to speaker pipeline: 
aplay -Dhw:0,2 -f dat -c 2 -i -vvv /dev/zero

**Reproduction Rate**
100%

**Reproduced with**
Latest daily sof-dev kernel
Latest SOF main
topology:  sof-mtl-rt711-l0-rt1316-l23-rt714-l1.tplg
Platform: {MTL SDW} with external codecs on GEN4 AIOC (sh-mtlp-rvp-sdw-09, not in CI farm)
-	Headset: ALC711 (headset) on link 0, 
-	Speaker: ALC1316 x2 on link 2 & 3 
-	DMIC: ALC714 on link1.  

kernel options snd_soc_sof_sdw quirk=2

The first bad commit is: e424b874a477617d080f368ae67b46d19b7e6d4d
![image](https://github.com/thesofproject/sof/assets/8703742/4224a916-a5e2-46e5-b910-71d538ccbf14)

**Comment by mengdonglin** [comment]
@ranj063 Can you help check this regression?

**Comment by ranj063** [comment]
> @ranj063 Can you help check this regression?

@mengdonglin I think I know the issue. We're missing the src buffer update in the case of multie endpoint playback. I've given @btian1 the change to try out and check if it fixes the problem

**Comment by lgirdwood** [comment]
> it fixed this issue, please check with below PR, @aiChaoSONG @bardliao @ranj063 #7987

@aiChaoSONG Now merged #7987.

**Comment by aiChaoSONG** [comment]
Verified fix #7987, close.
